src/meson.build

```diff
@@ -789,15 +791,97 @@ endif
 
 have_gobject = conf.get('HAVE_GOBJECT', 0) == 1
 
+# This code (especially PACKAGE_INIT) kept similar to what CMake's own
+# configure_package_config_file() generates, see
+# https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#command:configure_package_config_file
 cmake_config = configuration_data()
-cmake_config.set('libdir', get_option('prefix') / get_option('libdir'))
-cmake_config.set('includedir', get_option('prefix') / get_option('includedir'))
-cmake_config.set('HB_LIBTOOL_VERSION_INFO', hb_libtool_version_info)
-cmake_config.set('have_gobject', '@0@'.format(have_gobject))
+
+# Equivalent to configure_package_config_file(INSTALL_DESTINATION ...) (see link
+# above).
+# In Nixpkgs, the Harfbuzz package is installed into two Nix store paths, "out"
+# and "dev", where "out" contains libraries only (i.e. lib/libharfbuzz.so) and
+# "dev" contains development files, i.e. include and lib/cmake. If a CMake file
+# is installed to "out", Nixpkgs will move it to "dev". Since we need to figure
+# out relative install paths here, we do need to know the final path of our
+# CMake files to calculate the correct relative paths.
+# Of course, this still defaults to $libdir/cmake if unset.
+cmake_package_install_dir = get_option('cmakepackagedir')
+
+if cmake_package_install_dir == ''
+  cmake_package_install_dir = get_option('libdir') / 'cmake'
+endif
+
+cmake_config_dir = cmake_package_install_dir / 'harfbuzz'
+
+# TODO: Replace these with
+# fs.relative_to(get_option('prefix') / cmake_config_dir, get_option('prefix'))
+# etc. once implemented
+# See https://github.com/mesonbuild/meson/pull/12087
+relative_to = find_program('relative_to.py')
```

Comment:
PR#12087 of meson is merged, so now this should be replaced with `fs.relative_to). And the temporary solution `relative_to.py` can be pruned from the commit as it becomes depreciated.

Reply:
Ah, nice! It'll be there in Meson 1.3.0 which isn't released yet, but I can update the PR already.